code/datums/ship.dm

```diff
@@ -268,27 +267,154 @@ var/next_ship_id
 /datum/component/weapon/random/special
 	name = "special mount"
 	cname = "s_weapon"
+	fire_rate = 300
 
 	possible_weapons = list(/datum/ship_attack/ion,/datum/ship_attack/stun_bomb,/datum/ship_attack/flame_bomb)
 
 /datum/component/weapon/random/memegun
 	name = "meme weapon"
 	cname = "meme_weapon"
+	fire_rate = 100
 
 	possible_weapons = list(/datum/ship_attack/slipstorm,/datum/ship_attack/honkerblaster,/datum/ship_attack/bananabomb)
 
+
+		//Phase Cannons
+	/datum/component/slowweapon
+		name = "slow phase cannon"
+		cname = "slow_weapon"
+
+		flags = SHIP_WEAPONS
+		attack_data = /datum/ship_attack/laser
+		var/fire_rate = 300
+
+		alt_image = "weapon"
+
+	/datum/component/fastweapon
+		name = "fast phase cannon"
+		cname = "fast_weapon"
+
+		flags = SHIP_WEAPONS
+		attack_data = /datum/ship_attack/laser
+		var/fire_rate = 100
+
+		alt_image = "weapon"
+
+
+		//MAC Cannons
+/datum/component/weapon/mac_cannon
+	name = "MAC cannon"
+	cname = "mac_cannon"
+	fire_rate = 400
+
+	attack_data = /datum/ship_attack/ballistic
+
+/datum/component/weapon/slow_mac_cannon
+	name = "slow MAC cannon"
+	cname = "slow_mac_cannon"
+	fire_rate = 600
+
+	attack_data = /datum/ship_attack/ballistic
+
+/datum/component/weapon/fast_mac_cannon
+	name = "fast MAC cannon"
+	cname = "fast_mac_cannon"
+	fire_rate = 200
+
+	attack_data = /datum/ship_attack/ballistic
+
+		//Ion Cannons
 /datum/component/weapon/ion
 	name = "ion cannon"
 	cname = "ion_weapon"
+	fire_rate = 300
+
+	attack_data = /datum/ship_attack/ion
+
+/datum/component/weapon/slow_ion
+	name = "slow ion cannon"
+	cname = "slow_ion_weapon"
+	fire_rate = 450
+
+	attack_data = /datum/ship_attack/ion
+
+/datum/component/weapon/fast_ion
+	name = "fast ion cannon"
+	cname = "fast_ion_weapon"
+	fire_rate = 150
 
 	attack_data = /datum/ship_attack/ion
 
+
+		//Firebombs
+	datum/component/weapon/firebomb
+	name = "firebomber"
+	cname = "ion_firebomber"
+	fire_rate = 300
+
+	attack_data = /datum/ship_attack/flame_bomb
+
+/datum/component/weapon/slow_firebomb
+	name = "slow firebomber"
+	cname = "slow_firebomber"
+	fire_rate = 450
+
+	attack_data = /datum/ship_attack/flame_bomb
+
+/datum/component/weapon/fast_firebomb
+	name = "fast firebomber"
+	cname = "fast_firebomber"
+	fire_rate = 150
+
+	attack_data = /datum/ship_attack/flame_bomb
+
+
+			//Stun Bombs
+/datum/component/weapon/stunbomb
+	name = "stunbomber"
+	cname = "stunbomber"
+	fire_rate = 300
+
+	attack_data = /datum/ship_attack/stunbomb
```

Comment:
yeah I saw. I'm currently doing a commit it'll be up in about 30 minutes with a lot of extra stuff. Thanks for pointing it out, though :)